### PR TITLE
Adding ChainID Currency Symbol Map

### PR DESCRIPTION
**Pull request by vthomas13**
## **Description**
We want a ChainID: Ticker Map for future validation on some of our more commonly used networks. This PR adds that mapping constant, changes Optimism's "OP" ticker to "ETH" and fixes a resulting unit test failure.

## **Related issues**

Fixes: [#1708
](https://github.com/MetaMask/MetaMask-planning/issues/1708)
## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [X] I've included screenshots/recordings if applicable
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [X] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: shared/constants/network.ts [hunk 1]
```diff
@@ -343,6 +347,19 @@ export const NETWORK_TO_NAME_MAP = {
   [CHAIN_IDS.LOCALHOST]: LOCALHOST_DISPLAY_NAME,
 } as const;
 
+export const CHAIN_ID_TO_CURRENCY_SYMBOL_MAP = {
```
Comment (darkwing) [inline shared/constants/network.ts:350]: Where is this used?

Comment (vthomas13) [reply shared/constants/network.ts:350]: The usage isn't fully refined/discussed yet, but I think the future intention is that it should be used as a shortlist to query against so we can avoid calling chainlist API for 90% of our users who have that setting turned on

File: ui/hooks/useAccountTotalFiatBalance.test.js [hunk 2]
```diff
@@ -131,7 +131,7 @@ describe('useAccountTotalFiatBalance', () => {
       orderedTokenList: [
         {
           fiatBalance: '1.85',
-          iconUrl: './images/eth_logo.png',
+          iconUrl: './images/optimism.svg',
```
[the remark this answers is not in the document]
Comment (danjm) [reply ui/hooks/useAccountTotalFiatBalance.test.js:134]: the image should depend on the chain id and not the token symbol, but I also am curious why this test had to change

[the remark this answers is not in the document]
Comment (vthomas13) [reply ui/hooks/useAccountTotalFiatBalance.test.js]: This test failure was caused by a map collision, so I switched to a chainID based map in the latest commit [181f89c](https://github.com/MetaMask/metamask-extension/pull/22292/commits/181f89cf67021f0b787379d7e796621c234a5dd3)

**Review by pedronfigueiredo — APPROVED** [review]

**Review by danjm — APPROVED** [review]

**Pull request merged**